### DISCUSSION TITLE
Want to get this working for a new IPTV provider (not on the list) and UDM-SE - but it fails

### DISCUSSION THREAD

**Discussion by horolux** [fabianishere/udm-iptv#75] [Q&A]
Hi,

I'm actually trying to set this up on an UDM-SE, latest firmware and network application version.  The first install failed completely. I deleted everything and after downloading from scratch, the installation process worked, I had a kind of "graphic" interface to enter the configuration data. But at the end, I did not work at all.  So I removed everything.

Today, I'm starting over again, and the installation process is different from the 2 first ones:  this time I'm asked for the configuration data directly in the console, so not via this sort of "graphical" interface. Just at the end I've got asked about the IPTV VLAN via the graphical interface, but I already answered the question.

At the end:  it does not work.  

By the way, the WAN Ranges in the debug output are not the ones I've entered during the setup. Strange.

```
Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth8
WAN Ranges: 213.75.0.0/16 217.166.0.0/16
WAN VLAN: 0 (dev iptv)
WAN DHCP options: -O staticroutes -V IPTV_RG
LAN Interfaces: br0
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: false
=== IP Link and Route ===
=== Service Logs ===
Feb 21 11:57:05 UDM-SE systemd[1]: Stopped IPTV support for the UniFi Dream Machine.
Feb 21 11:57:05 UDM-SE systemd[1]: udm-iptv.service: Consumed 85ms CPU time.
Feb 21 11:57:05 UDM-SE systemd[1]: Started IPTV support for the UniFi Dream Machine.
Feb 21 11:57:05 UDM-SE udm-iptv[11122]: NATing IPTV network ranges (if necessary)
Feb 21 11:57:05 UDM-SE udm-iptv[11122]: Setting up igmpproxy
Feb 21 11:57:05 UDM-SE udm-iptv[11122]: Starting igmpproxy
Feb 21 11:57:05 UDM-SE igmpproxy[11122]: There must be at least 1 Vif as upstream.
Feb 21 11:57:05 UDM-SE systemd[1]: udm-iptv.service: Main process exited, code=exited, status=255/EXCEPTION
Feb 21 11:57:05 UDM-SE systemd[1]: udm-iptv.service: Failed with result 'exit-code'.
Feb 21 11:57:05 UDM-SE systemd[1]: udm-iptv.service: Consumed 90ms CPU time.

```

**Comment by horolux** [comment]
this discussion can be deleted as I've opened an issue.

**Reply by fabianishere** [reply]
Hi, since your issue is related to configuration and not to a issue with the project, I prefer having this discussion here.

Could you share which ISP you are connected to?

**Reply by horolux** [reply]
Hi,

My ISP is Visual Online, but I think that this will not help you ;-)  

Before moving to Unifi I had a Fritzbox connected to my ISP over VDSL2. The IPTV Decoder was connected to a small Netgear Switch which was connected to the FB. I was working fine. Now I have a Draytek Vigor 167 Modem (no router!!) connected to the UDM-SE (WAN is configured as PPPoE) and my IPTV is not working anymore. 

I was starting to analyze the FB backup file and found 3 VLANs:  Internet on 35, VOIP on 39 and IPTV on 37. Discussed with some technical people I was told that this is propably due to an multi-vc DSL connection and that I should ask to get an mono-vc connection. This has been done, do normally I have not to deal anymore with multiples VLAN IDs on the provider side.  I only have to indicate VLAN ID 35 to get my Internet connection working (otherwise I don't get an WAN IP Address on my UDM-SE).

So I have eth8 and eth8.35 but also ppp0.
eth8 as well as eth8.35 don't have IP addresses assigned.
ppp0 has my official IP address from my ISP (but I cannot enter/choose pppo in the installation dialog)

I think that the installation has gone wrong as there is also no device iptv defined while the configuration file is reference to such a device. 


```
root@UDM-SE:/etc# cat udm-iptv.conf
## Example configuration for udmp-iptv ##
# Interface on which IPTV traffic enters the router
IPTV_WAN_INTERFACE="eth8"
# ID of VLAN which carries IPTV traffic (use 0 if no VLAN is used)
IPTV_WAN_VLAN="0"
# Name of the IPTV VLAN interface
**IPTV_WAN_VLAN_INTERFACE="iptv"**
# IP ranges from which the IPTV traffic originates (separated by spaces)
IPTV_WAN_RANGES="224.0.0.0/8"
# DHCP options to send when requesting an IP address
IPTV_WAN_DHCP_OPTIONS=""
# LAN interfaces on which IPTV should be made available
IPTV_LAN_INTERFACES="br0"
# Disable quickleave for igmpproxy
IPTV_IGMPPROXY_DISABLE_QUICKLEAVE="false"
# Enable debugging for igmpproxy
IPTV_IGMPPROXY_DEBUG="true"root@UDM-SE:/etc# 
```

**Reply by horolux** [reply]
What do I need to provide? We can also have a zoom session and I can share the screen with you - just let me know.

**Reply by fabianishere** [reply]
Do you now if the IPTV traffic is carried over the PPPoE tunnel or outside it?

**Reply by fabianishere** [reply]
If not, you should set `IPTV_WAN_VLAN` to `37`.

**Reply by horolux** [reply]
To be honest: no. 

If the decoder box is connected over WLAN it is working (without unifi-iptv). The WLAN Accesspoints are connected to a switch behind the UDM-SE.
If I connect the decoder to the LAN it is not working but if I hit "pause" for 2 seconds and then "play" it starts and keeps working (as it probably changes to Unicast?

**Reply by horolux** [reply]
I already tried to set it to 37, but I'm pretty sure that the udm-iptv/igmpproxy setup/installation is not correct, so we shall first make sure that everything is correctly installed, up and running before trying to change VLAN ids etc ;-). 

How can I check / what shall I provide to make sure the installation is fine and everything is up and running as forseen?

**Reply by fabianishere** [reply]
What is your reason for thinking the installation is not correct?

**Reply by horolux** [reply]
There are several reasons:

1.  I started 3 times from scratch, each time the installation took another path, was not 2 times the same behavior
2. After the latest installation the configuration file did not contain what I've entered during the installation, I had to run
    the reconfiguration in order to get it right
3. ifconfig does not show an iptv device. (which was present after one of the previous installations)
4. igmpproxy seems not present (maybe this is normal, but it seems not up and running)

So me feeling is that something is messed up.  Also that the last installation stated that there is no dialog-like program while it has been used just one installation before makes the feeling not better ;-)

I prefer analyzing this in a structured way, in this case this means for me: make sure that everything has been installed as it should and that everything is up and running as it should.  The second step will then be finetuning IP addresses or VLAN Ids ;-)  we will loose time searching if the basement is not correct ;-)

[Terminal Saved Output.txt](https://github.com/fabianishere/udm-iptv/files/8109565/Terminal.Saved.Output.txt)

**Reply by fabianishere** [reply]
1. Could happen when the configuration of `udm-iptv` is started before `dialog`, though this is only an inconvenience, I have updated the installation instructions so this won't happen again.
2. That should normally not happen. If you can reproduce the issue, please file a bug report.
3. This happens when you set `IPTV_WAN_VLAN` to zero.
4. This is expected behavior. `udm-iptv` does not alter any settings from `igmpproxy`; it invokes it directly.

I don't think the installation is the problem in your case.

**Reply by horolux** [reply]
OK, I changed the /etc/udm-iptv.conf and put something different than zero to IPTV_WAN_VLAN and executed the restart command from the readme page. Still no iptv device in ifconfig. 

I must say that I don't really know why we're trying here to get another IP from the ISP?  We have an IP from the ISP which is related to the WAN Port (ppp0).  Why shall there be another one assigned? My decoder box as an IP address on my internal network, and each time when I switch channels it does a LEAVE GROUP or GROUP REPORT which is send to 224.0.0.2 and groups leaved or joined are 224.11.2.x  I see also IGMP v2 in the packet (was playing with wireshark).   

My understanding is that these multicast packages are not routed outside my LAN to WAN and there's where IGMP Proxy is coming into the game?

**Reply by horolux** [reply]
I tried another reconfiguration:

```
Feb 21 15:22:44 UDM-SE systemd[1]: Started IPTV support for the UniFi Dream Machine.
Feb 21 15:22:44 UDM-SE udm-iptv[16687]: Device iptv already exists.. Deleting device
Feb 21 15:22:44 UDM-SE udm-iptv[16687]: RTNETLINK answers: Cannot assign requested address
Feb 21 15:22:44 UDM-SE udm-iptv[16687]: Obtaining IP address for VLAN interface
Feb 21 15:22:44 UDM-SE udm-iptv[16687]: udhcpc (v1.22.1) started
Feb 21 15:22:45 UDM-SE udm-iptv[16687]: Sending discover...
Feb 21 15:22:45 UDM-SE udm-iptv[16687]: udhcpc: sendto: Network is down
Feb 21 15:22:45 UDM-SE udm-iptv[16687]: Read error: Network is down, reopening socket
Feb 21 15:22:48 UDM-SE udm-iptv[16687]: Sending discover...
Feb 21 15:22:48 UDM-SE udm-iptv[16687]: udhcpc: sendto: Network is down
Feb 21 15:22:48 UDM-SE udm-iptv[16687]: Read error: Network is down, reopening socket
Feb 21 15:22:51 UDM-SE udm-iptv[16687]: Sending discover...
Feb 21 15:22:51 UDM-SE udm-iptv[16687]: udhcpc: sendto: Network is down
Feb 21 15:22:51 UDM-SE udm-iptv[16687]: Read error: Network is down, reopening socket
Feb 21 15:22:54 UDM-SE udm-iptv[16687]: udhcpc failed to get a DHCP lease
Feb 21 15:22:54 UDM-SE udm-iptv[16687]: No lease, forking to background
Feb 21 15:22:54 UDM-SE udm-iptv[16687]: NATing IPTV network ranges (if necessary)
Feb 21 15:22:54 UDM-SE udm-iptv[16687]: Setting up igmpproxy
Feb 21 15:22:54 UDM-SE udm-iptv[16687]: Starting igmpproxy
Feb 21 15:22:54 UDM-SE udm-iptv[16687]: adding VIF, Ix 0 Fl 0x0 IP 0x010a000a br0, Threshold: 1, Ratelimit: 0
Feb 21 15:22:54 UDM-SE udm-iptv[16687]: There must be at least 1 Vif as upstream.
Feb 21 15:22:54 UDM-SE systemd[1]: udm-iptv.service: Main process exited, code=exited, status=255/EXCEPTION
Feb 21 15:22:55 UDM-SE systemd[1]: udm-iptv.service: Failed with result 'exit-code'.
Feb 21 15:22:55 UDM-SE systemd[1]: udm-iptv.service: Consumed 103ms CPU time.
Feb 21 15:23:00 UDM-SE systemd[1]: udm-iptv.service: Service RestartSec=5s expired, scheduling restart.
Feb 21 15:23:00 UDM-SE systemd[1]: udm-iptv.service: Scheduled restart job, restart counter is at 43.
Feb 21 15:23:00 UDM-SE systemd[1]: Stopped IPTV support for the UniFi Dream Machine.
```

**Reply by fabianishere** [reply]
Since IPTV traffic is carried over a separate VLAN (37), you'll need a IP address to be able to communicate over this VLAN.

Please set `IPTV_WAN_VLAN` to 37 and share the output of `udm-iptv-diag`.

**Reply by horolux** [reply]
Ok, I will try this and send you the output

But remember:  the separate VLAN was on the multi-vc DSL Line, I was told that with the mono-vc DSL everything should be carries over the same VLAN 35.

**Reply by fabianishere** [reply]
In that case, `IPTV_WAN_VLAN` should be set to zero and `IPTV_WAN_INTERFACE` to `ppp0`.

You should also think about what appropriate values for `IPTV_WAN_RANGES` are. Perhaps your ISP has documented this. Otherwise, you may need to create a packet trace of your network.

**Reply by horolux** [reply]
Hi,

Seems that we are 1 step closer,  the output looks a lot better:

```
root@UDM-SE:/etc# udm-iptv-diag
Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth8
WAN Ranges: 224.0.0.0/8
WAN VLAN: 37 (dev iptv)
WAN DHCP options: -O staticroutes -V IPTV_RG
LAN Interfaces: br0
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
=== Service Logs ===
Feb 21 15:38:11 UDM-SE systemd[1]: udm-iptv.service: Scheduled restart job, restart counter is at 4.
Feb 21 15:38:11 UDM-SE systemd[1]: Stopped IPTV support for the UniFi Dream Machine.
Feb 21 15:38:11 UDM-SE systemd[1]: udm-iptv.service: Consumed 100ms CPU time.
Feb 21 15:38:11 UDM-SE systemd[1]: Started IPTV support for the UniFi Dream Machine.
Feb 21 15:38:11 UDM-SE udm-iptv[8519]: Device iptv already exists.. Deleting device
Feb 21 15:38:11 UDM-SE udm-iptv[8519]: Obtaining IP address for VLAN interface
Feb 21 15:38:11 UDM-SE udm-iptv[8519]: udhcpc (v1.22.1) started
Feb 21 15:38:11 UDM-SE udm-iptv[8519]: Sending discover...
Feb 21 15:38:14 UDM-SE udm-iptv[8519]: Sending discover...
Feb 21 15:38:18 UDM-SE udm-iptv[8519]: Sending discover...
root@UDM-SE:/etc# 

```

Now, the question is:  why is my IPTV still not working? There must be a next step to verifiy ?

**Reply by horolux** [reply]
I was too fast:

```
root@UDM-SE:/etc# udm-iptv-diag
Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth8
WAN Ranges: 224.0.0.0/8
WAN VLAN: 37 (dev iptv)
WAN DHCP options: -O staticroutes -V IPTV_RG
LAN Interfaces: br0
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
=== Service Logs ===
Feb 21 15:43:15 UDM-SE udm-iptv[23598]: udhcpc failed to get a DHCP lease
Feb 21 15:43:15 UDM-SE udm-iptv[23598]: No lease, forking to background
Feb 21 15:43:15 UDM-SE udm-iptv[23598]: NATing IPTV network ranges (if necessary)
Feb 21 15:43:15 UDM-SE udm-iptv[23598]: Setting up igmpproxy
Feb 21 15:43:15 UDM-SE udm-iptv[23598]: Starting igmpproxy
Feb 21 15:43:15 UDM-SE udm-iptv[23598]: adding VIF, Ix 0 Fl 0x0 IP 0x010a000a br0, Threshold: 1, Ratelimit: 0
Feb 21 15:43:15 UDM-SE udm-iptv[23598]: There must be at least 1 Vif as upstream.
Feb 21 15:43:15 UDM-SE systemd[1]: udm-iptv.service: Main process exited, code=exited, status=255/EXCEPTION
Feb 21 15:43:15 UDM-SE systemd[1]: udm-iptv.service: Failed with result 'exit-code'.
Feb 21 15:43:15 UDM-SE systemd[1]: udm-iptv.service: Consumed 90ms CPU time.
root@UDM-SE:/etc# 
```

**Reply by horolux** [reply]
by the way, the configuration dialog doesn't let me configure ppp0 it prefills the options with WAN1 RJ45 or WAN2. That's all. I can change the config file manually, but as stated in the readme, it will not persist. 

If I choose "No" when asked if the IPTV enters via a specifc VLAN the next screen gives me options to choose from, but ppp0 is missing in the list


I see: eth8 ; eth8.35 and iptv (must be from udm-iptv)

**Reply by fabianishere** [reply]
That is indeed an issue in the current version of the package (that you cannot select `ppp0`). Just make the changes to `/etc/udm-iptv.conf` manually (the changes will actually persist in the latest version, I have updated the guide to reflect that) and restart the service.

**Reply by horolux** [reply]
ok, do I also have do out IPTV_WAN_VLAN_INTERFACE into comments ?  As no VLAN 0 no VLAN shall be used?

**Reply by fabianishere** [reply]
You don't have to. It won't create an interface if `IPTV_WAN_VLAN` is set to zero.

**Comment by horolux** [comment]
Looks a lot better:

```
Please share the following output with the developers:
=== Configuration ===
WAN Interface: ppp0
WAN Ranges: 224.0.0.0/8
WAN VLAN: 0 (dev iptv)
WAN DHCP options: -O staticroutes -V IPTV_RG
LAN Interfaces: br0
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
31: ppp0: <POINTOPOINT,MULTICAST,NOARP,UP,LOWER_UP> mtu 1492 qdisc pfifo_fast state UNKNOWN group default qlen 3
    inet xxxxxxxxxxxxxxxx peer yyyyyyyyyyyyyyyy scope global ppp0
       valid_lft forever preferred_lft forever
xxxxxxxxxxx proto kernel scope link src yyyyyyyyyyyyyyyyy 
=== Service Logs ===
Feb 21 15:58:54 UDM-SE udm-iptv[23035]: Updated route entry for 239.254.127.63 on VIF #0
Feb 21 15:58:55 UDM-SE udm-iptv[23035]: The IGMP message was local multicast. Ignoring.
Feb 21 15:58:57 UDM-SE udm-iptv[23035]: The IGMP message was local multicast. Ignoring.
Feb 21 15:58:59 UDM-SE udm-iptv[23035]: RECV Leave message      from 10.0.10.141     to 224.0.0.2
Feb 21 15:58:59 UDM-SE udm-iptv[23035]: Leaving group 224.11.2.3 on interface ppp0
Feb 21 15:58:59 UDM-SE udm-iptv[23035]: RECV Membership query   from 10.0.10.1       to 224.11.2.3
Feb 21 15:58:59 UDM-SE udm-iptv[23035]: RECV V2 member report   from 10.0.10.141     to 224.11.2.4
Feb 21 15:58:59 UDM-SE udm-iptv[23035]: Inserted route table entry for 224.11.2.4 on VIF #0
Feb 21 15:58:59 UDM-SE udm-iptv[23035]: Joining group 224.11.2.4 on interface ppp0
Feb 21 15:59:02 UDM-SE udm-iptv[23035]: The IGMP message was local multicast. Ignoring.

```

**Comment by horolux** [comment]
but I have still no image...IPTV not working

When I switch channels and look at the udm-iptv-diag I see the new group member reports (leaving / joining) but nothing happens on the TV ;-)

**Reply by horolux** [reply]
Is there a log where I can see more as the few lines from udm-iptv-diag ? 

the problem is that my IPTV provider is not providing IP addresses, switching from one channel to another is clearly showing the multicast groups joined / unjoined but is that all?  

I mean without a TV channel which is really starting, how shall I see something with wireshark on ppp0 ? ;-)

**Reply by fabianishere** [reply]
It is possibly failing when trying to connect to some address (which we can see in Wireshark).

If you want more logs: `journalctl -u udm-iptv`

**Reply by horolux** [reply]
I'm lost...don't get it working and dont' know on what to look else.  As I'm not an expert wireshark user nor an expert network engineer (studies are toooo far away lol), I'm "finding back" my knowledge step by step, but I wanna get this working (and then add the info here to the configuration table). I just need some help.

**Reply by reinierjh** [reply]
you could try to change the WAN range to 0.0.0.0/0
this allows all IGMP traffic from everywhere, and masquerades (NAT) al traffic to everywhere when it exits the IPTV interface.
And then looking at for example a "tcpdump -i iptv" command try to find wich addresses you really need.

**Reply by horolux** [reply]
Hi, thanks for the help, I give this a try. So far I tried with 0.0.0.0/8 added. I'll come back here with the results.

**Reply by horolux** [reply]
Hi,

So I tried this, still not working. I've attached an extract of the dump with only eth.dst contains 01:00:5e (=all multicast traffic). 

224.11.2.x seem to be my channels as you see that I switched between channels and there are corresponding group leaves/joins. 

But what I don't see is if this traffic is leaving leaving my network or not. The source 10.x.x.75 is the IPTV box.  

[TCPDUMP_IPTV_MC.xlsx](https://github.com/fabianishere/udm-iptv/files/8119524/TCPDUMP_IPTV_MC.xlsx)

**Reply by horolux** [reply]
Hi, still not working :-( ....
